From 967bccd6ad8f5b83bdd4892f660626da412e3555 Mon Sep 17 00:00:00 2001
From: He Junyan <junyan.he@intel.com>
Date: Thu, 25 May 2023 14:48:11 +0800
Subject: [PATCH 76/77] glupload: Fix a memory leak point in
 gst_egl_image_cache_unref()

The image cache itself should be freed when ref_count is 0.

Part-of: <https://gitlab.freedesktop.org/gstreamer/gstreamer/-/merge_requests/4723>
---
 gst-libs/gst/gl/gstglupload.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/gst-libs/gst/gl/gstglupload.c b/gst-libs/gst/gl/gstglupload.c
index 7a8c3693..4c0be070 100644
--- a/gst-libs/gst/gl/gstglupload.c
+++ b/gst-libs/gst/gl/gstglupload.c
@@ -615,6 +615,7 @@ gst_egl_image_cache_unref (GstEGLImageCache * cache)
   if (g_atomic_int_dec_and_test (&cache->ref_count)) {
     g_hash_table_unref (cache->hash_table);
     g_mutex_clear (&cache->lock);
+    g_free (cache);
   }
 }
 
-- 
2.40.1

